src/library/scala/collection/Factory.scala

```diff
@@ -61,7 +61,7 @@ object Factory {
   private class ArrayFactory[A: ClassTag] extends Factory[A, Array[A]] with Serializable {
     def fromSpecific(it: IterableOnce[A]): Array[A] = {
       val b = newBuilder
-      b.sizeHint(scala.math.max(0, it.knownSize))
+      b.sizeHint(it, delta = 0)
```

Comment:
hint at 0 is no-op because capacity cannot be less than 0. Empty array is allocated on result, if nothing was added.

(The difference is ArrayBuffer has a default capacity.)